core/src/main/scala/org/apache/spark/deploy/Client.scala

```diff
@@ -17,33 +17,38 @@
 
 package org.apache.spark.deploy
 
-import scala.concurrent._
+import scala.concurrent.ExecutionContext
+import scala.util.{Failure, Success}
 
-import akka.actor._
-import akka.pattern.ask
-import akka.remote.{AssociationErrorEvent, DisassociatedEvent, RemotingLifecycleEvent}
 import org.apache.log4j.{Level, Logger}
 
+import org.apache.spark.rpc.{RpcEndpointRef, RpcAddress, RpcEnv, ThreadSafeRpcEndpoint}
 import org.apache.spark.{Logging, SecurityManager, SparkConf}
 import org.apache.spark.deploy.DeployMessages._
 import org.apache.spark.deploy.master.{DriverState, Master}
-import org.apache.spark.util.{ActorLogReceive, AkkaUtils, Utils}
+import org.apache.spark.util.Utils
 
 /**
  * Proxy that relays messages to the driver.
  */
-private class ClientActor(driverArgs: ClientArguments, conf: SparkConf)
-  extends Actor with ActorLogReceive with Logging {
-
-  var masterActor: ActorSelection = _
-  val timeout = AkkaUtils.askTimeout(conf)
-
-  override def preStart(): Unit = {
-    masterActor = context.actorSelection(
-      Master.toAkkaUrl(driverArgs.master, AkkaUtils.protocol(context.system)))
-
-    context.system.eventStream.subscribe(self, classOf[RemotingLifecycleEvent])
+private class ClientEndpoint(
+    override val rpcEnv: RpcEnv,
+    driverArgs: ClientArguments,
+    masterEndpoint: RpcEndpointRef,
+    conf: SparkConf)
+  extends ThreadSafeRpcEndpoint with Logging {
+
+  private val forwardMessageThread = Utils.newDaemonFixedThreadPool(1, "client-forward-message")
+  private implicit val forwardMessageExecutionContext =
+    ExecutionContext.fromExecutor(forwardMessageThread,
+      t => t match {
+        case ie: InterruptedException => // Exit normally
+        case e =>
+          e.printStackTrace()
+          System.exit(-1)
```

Comment:
I just followed other places in `Client` because `Client` is deprecated. However, I should have used correct ways in the new codes. Now I changed to use `SparkExitCode.UNCAUGHT_EXCEPTION`. Maybe I should also fix other `-1`s in this file?